examples/hovercard-shadow-dom/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { Fragment, useState } from "react";
+import type { FunctionComponent, HTMLProps } from "react";
+import * as Ariakit from "@ariakit/react";
+import root from "react-shadow";
+
+const ShadowRootHost = root.div as FunctionComponent<HTMLProps<HTMLDivElement>>;
+
+export default function Example() {
+  const [count, setCount] = useState(0);
+  return (
+    <Fragment>
+      <span>Opened {count} times</span>
+      <ShadowRootHost>
+        <Ariakit.HovercardProvider
+          setOpen={(open) => {
+            if (open) setCount(count + 1);
+
+            return open;
```

Comment:
Returning is not necessary.
```suggestion
            if (open) setCount(count + 1);
```